lib/Importer.php

```diff
@@ -211,7 +211,8 @@ private function _addCertificateMetaData(&$parsedMetaData)
 
         foreach ($certKeys as $certKey) {
             if (isset($parsedMetaData[$certKey . 'X509Certificate']) &&
-                (!isset($parsedMetaData[$certKey . 'encryption']) ||
+                ( (isset($parsedMetaData[$certKey . 'signing']) && $parsedMetaData[$certKey . 'signing']) ||
+                    !isset($parsedMetaData[$certKey . 'encryption']) ||
```

Comment:
Actually I think it is that simple, because those if conditions are simply guarding that we don't use encryption keys (not encryption and signing keys) if we are not using encryption.